# Notebook Instructions
<i>You can run the notebook document sequentially (one cell a time) by pressing <b> shift + enter</b>. While a cell is running, a [*] will display on the left. When it has been run, a number will display indicating the order in which it was run in the notebook [8].</i>

<i>Enter edit mode by pressing <b>`Enter`</b> or using the mouse to click on a cell's editor area. Edit mode is indicated by a green cell border and a prompt showing in the editor area.</i>

# Call price using Delta and Gamma

In this notebook, we will calculate approximate price of call option using Delta and Gamma. The formula to compute the call option price using Taylor series expansion is given as:

<i><b>Price of a call option = Initial Option Price + Delta x (change in the stock price) + 0.5 x Gamma x (change in the stock price)^2</b></i>



# Delta and Gamma when stock price is at 100

In [1]:
# Import Library
import mibian
from tabulate import tabulate

# Define parameters for call option contract
stock_price = 100
strike_price = 100
interest_rate = 0
days_to_expiry = 40
volatility = 38.09

# Compute greeks
greeks_stock_price_100 = mibian.BS([stock_price, strike_price, interest_rate, days_to_expiry], \
                   volatility=volatility)

# Print Delta and Gamma
print tabulate(
        [             
            ['Stock Price', 100],
            ['Call Price', greeks_stock_price_100.callPrice],
            ['Delta', greeks_stock_price_100.callDelta],
            ['Gamma', greeks_stock_price_100.gamma]            
        ], 
        tablefmt='orgtbl')

| Stock Price | 100         |
| Call Price  |   5.02709   |
| Delta       |   0.525135  |
| Gamma       |   0.0315757 |


# Call price when stock is at 101 using Delta and Gamma

In [2]:
# Change in stock price is Rs. 1 from 100 to 101
call_price_at_101 = greeks_stock_price_100.callPrice \
                  + greeks_stock_price_100.callDelta * (101 - 100) \
                  + 0.5 * greeks_stock_price_100.gamma * (101 - 100)**2
print call_price_at_101

5.56801680203


# Actual call price when stock is at 101

In [3]:
stock_price = 101
actual_call_price_at_101 = mibian.BS([stock_price, strike_price, interest_rate, days_to_expiry], \
                   volatility=volatility).callPrice

print actual_call_price_at_101

5.56793020593


Thus the call price computed using Delta and Gamma of 5.5680 is pretty much close to the actual price of 5.5679. The call price is not equal to actual price due to the fact that there are higher order Greeks than Gamma. But as seen here they do not have much of an impact.